SysBot.Pokemon.YouTube/YouTubeBot.cs

```diff
@@ -0,0 +1,185 @@
+﻿using Google.Apis.YouTube.v3.Data;
+using PKHeX.Core;
+using StreamingClient.Base.Util;
+using SysBot.Base;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YouTube.Base;
+using YouTube.Base.Clients;
+
+namespace SysBot.Pokemon.YouTube
+{
+    public class YouTubeBot
+    {
+        private static PokeTradeHub<PK8> Hub;
+        internal static TradeQueueInfo<PK8> Info => Hub.Queues.Info;
+        private ChatClient client;
+        private Channel channel;
+        internal static readonly List<YouTubeQueue> QueuePool = new List<YouTubeQueue>();
+        private readonly YouTubeSettings Settings;
+
+        public YouTubeBot(YouTubeSettings settings, PokeTradeHub<PK8> hub)
+        {
+            Hub = hub;
+            Settings = settings;
+            Logger.LogOccurred += Logger_LogOccurred;
+
+            try
+            {
+                Task.Run(async () =>
+                {
```

Comment:
Thread will NRE if connection/channel/client fail to connect... Not sure if a while true loop is appropriate. Await task delay -1 might be better.